code/modules/overmap/ships/computers/ship.dm

```diff
@@ -28,6 +77,38 @@ somewhere on that shuttle. Subtypes of these can be then used to perform ship ov
 	src.add_hiddenprint(user)
 	ui_interact(user)
 
+/obj/machinery/computer/ship/get_examine_text(mob/user, distance, is_adjacent, infix, suffix)
+	. = ..()
+	if(initial(hotwire_progress) != hotwire_progress)
+		if(hotwire_progress != 0)
+			. += SPAN_SUBTLE("The bottom panel appears open with wires hanging out. It can be repaired with additional cabling. <i>Current progress: [(hotwire_progress / initial(hotwire_progress)) * 100]%</i>")
+		else
+			. += SPAN_SUBTLE("The bottom panel appears open with wires hanging out. It can be repaired with additional cabling.")
+
+/obj/machinery/computer/ship/emag_act(var/remaining_charges, var/mob/user, var/hotwired = FALSE)
```

Comment:
Respect the parent proc signature
`/atom/proc/emag_act(var/remaining_charges, var/mob/user, var/emag_source)`
You are not keeping the `emag_source` var